src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -206,15 +206,29 @@ public function has($id)
             ) {
                 return true;
             }
+
+            if (isset($this->privates[$id])) {
```

Comment:
Reorganizing this code was done to match the way the code is organized in ``get``, so that we first try to find the service as is, and then do the lowercase conversion if needed. This way, we won't make a ``strtolower`` call when checking ``->has()`` for an existing service anymore when providing the lowercase string already.